django_irods/storage.py

```diff
@@ -372,7 +372,27 @@ def size(self, name):
         if "CAT_NO_ROWS_FOUND" in stdout:
             raise ValidationError("{} cannot be found in iRODS to retrieve "
                                   "file size".format(name))
-        return int(stdout)
+        return int(float(stdout))
+
+    def checksum(self, name):
+        # first force checksum (re)computation
+        self.session.run("ichksum", None, "-f", name)
+        # retrieve checksum using iquest
+        if '/' in name:
+            file_info = name.rsplit('/', 1)
+            obj_name = file_info[1]
+        else:
+            obj_name = name
+        qrystr = "SELECT DATA_CHECKSUM WHERE DATA_NAME = '{}'".format(obj_name)
```

Comment:
I see stuff like this and I think back to an internship I had where all I did was patch SQL injection vulnerabilities across the whole codebase.

Reply:
If we continue to go with icommands to retrieve these info, I don't see a way around this. I have created an issue here: https://github.com/hydroshare/hydroshare/issues/3744 to do an overhaul improvement later as a separate issue.